atomicapp/plugin.py

```diff
@@ -166,3 +167,10 @@ def getProvider(self, provider_key):
             if key == provider_key:
                 logger.debug("Found provider %s", provider)
                 return provider
+
+        # importing this on top of file leads to circular import
+        from atomicapp.nulecule.exceptions import NuleculeException
```

Comment:
So, as I had said earlier, let's raise an exception in ``nulecule/main.py`` when ``provider is None``.